src/rpc/misc.cpp

```diff
@@ -138,6 +139,63 @@ static UniValue createmultisig(const JSONRPCRequest& request)
     return result;
 }
 
+UniValue deriveaddress(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.empty() || request.params.size() > 1)
+        throw std::runtime_error(
+            "deriveaddress \"descriptor\"\n"
+            "\nDerives the address corresponding to an output descriptor.\n"
+            "Examples of output descriptors are:\n"
+            "    pkh(<pubkey>)                        P2PKH outputs for the given pubkey\n"
+            "    wpkh(<pubkey>)                       Native segwit P2PKH outputs for the given pubkey\n"
+            "    sh(multi(<n>,<pubkey>,<pubkey>,...)) P2SH-multisig outputs for the given threshold and pubkeys\n"
+            "    raw(<hex script>)                    Outputs whose scriptPubKey equals the specified hex scripts\n"
+            "\nIn the above, <pubkey> either refers to a fixed public key in hexadecimal notation, or to an xpub/xprv optionally followed by one\n"
+            "or more path elements separated by \"/\", where \"h\" represents a hardened child key.\n"
+            "For more information on output descriptors, see the documentation in the doc/descriptors.md file.\n"
+            "\nArguments:\n"
+            "1. \"descriptor\"                     (string, required) The descriptor\n"
+            "\nResult:\n"
+            "(string) The derived address\n"
+            "\nExamples:\n"
+            "\nNative segwit receive address\n"
+            + HelpExampleCli("deriveaddress", "\"wpkh([d34db33f/84h/0h/0h]xpub6DJ2dNUysrn5Vt36jH2KLBT2i1auw1tTSSomg8PhqNiUtx8QX2SvC9nrHu81fT41fvDUnhMjEzQgXnQjKEu3oaqMSzhSrHMxyyoEAmUHQbY/0/0)\"") +
+            ""
+        );
+
+    RPCTypeCheck(request.params, {UniValue::VSTR});
+
+    const std::string desc_str = request.params[0].get_str();
+
+    FlatSigningProvider provider;
+    auto desc = Parse(desc_str, provider);
+    if (!desc) {
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, strprintf("Invalid descriptor"));
+    }
+    if (desc->IsRange()){
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, strprintf("Descriptor range not supported"));
```

Comment:
It seems a weird anti-feature to me. I don't understand why you're adding extra complexity to the function just to prevent a range. If you only want one, you can still do so.

Reply:
@Sjors I assume that also includes allowing `combo` right? Sounds good 👍